Biocum/cherry pick

jaspnas opened the pull request.

andreedd committed:
add trakasserier

andreedd committed:
Name on signup required (#414)

* first and last name required on signup

* console email if debug

Theoabw committed:
Event redirect (#421)

* Event redirect

* Fix artifacts

Theoabw committed:
Event redirect (#421)

* Event redirect

* Fix artifacts

andreedd committed:
fix polls publish with none published bug

jaspnas committed:
Make event custom fields reorderable (#417)

* Make event custom fields reorderable

* Remake migration

jaspnas committed:
Fix server error on detail page if no custom registration fields are defined (#426)

JoarSabel committed:
Fixed hardcoded checks in archive index (#312)

jaspnas committed:
Add basic teapot functionality (#419)

jaspnas committed:
Return 404 instead of 500 if staticpage does not exist, clean staticpages module (#435)

Theoabw committed:
Fix/365 fix replace fontawesome kits (#449)

* replace font awesome with line awesome

* fixed a mixup

jaspnas committed:
Allow editing event signup fields (#434)

jaspnas committed:
Add random smoke effect (#420)

* Add random smoke effect

* Change trigger condition

jaspnas committed:
Add caching to rarely updated pages (#450)

* Add caching to rarely updated pages

5 min TTL for main page

1 minute TTL for index- and staticpages

* Make web container depend on redis

Tests will otherwise fail because of caching

jaspnas committed:
Move caching to templates (#454)

Theoabw committed:
Moved stuff (#459)

Moved manually run scripts from archive to scripts folder.

jaspnas committed:
Fix websockets (#464)

jaspnas committed:
feat: Celery mail (#433)

* Organize env file, add email options to example.env

* Replace core/.env with environment variables

* Send emails using celery workers

* Use celery for password reset emails

* Ficks stav fell

* Use logger instead of print in the email task

jaspnas committed:
Add export statements to example.env (#466)

Nyygard committed:
Open external links in different tab

jaspnas committed:
Fix password reset email sender (#471)

jaspnas committed:
Fix news returning 500 (#473)

jaspnas committed:
Make migrations

Theoabw reviewed (approved):
lgtm

theoabw commented:
![image](https://github.com/Datateknologerna-vid-Abo-Akademi/date-website/assets/12630429/fb983db8-7b20-4d7e-a813-2ee5d4014926)
dunno if this is only on my pc

jaspnas committed:
Calendar retry (#465)

* Replace calendar widget

Replace calendar widget with vanilla-calendar.js

Fix switching months reloading entire page

Add styling to new calendar

* Remove redundant files

Removed old calendar manager files

Refactor function that creates events JSON for calendar

Minor CSS changes

* Replace calendar widget

Replace calendar widget with vanilla-calendar.js

Fix switching months reloading entire page

Add styling to new calendar

* Remove redundant files

Removed old calendar manager files

Refactor function that creates events JSON for calendar

Minor CSS changes

* Re-add part of script removed during rebase

* Change calendar colors

---------

Co-authored-by: MirruK <mirek.kudinoff@abo.fi>

MirruK committed:
Calendar shows old events (max 1 month) (#476)

Change how the timestamp for the event
is rendered so that it accounts for client timezone

MirruK committed:
Remove race condition in script execution (#478)

jaspnas committed:
Fix cherry-pick bugs

Theoabw committed:
Custom change password template + button to userinfo (#457)

* Custom change password template + button to userinfo

* Fix template and button

* Fixed styles a bit

* clean up

Theoabw committed:
fixed calendar not in mobile view (#486)

* brought back necessary media queries

* fixed calendar not showing up in mobile view

* fixed calendar width and colors

Theoabw committed:
Update dependencies (#458)

jaspnas committed:
Fix prettyjson breaking in django 5.0 (#494)

jaspnas committed:
Fix 405 error on logout (#505)

jaspnas committed:
Fix 405 error on logout (#505)

Theoabw committed:
fix minutes < 10 (#510)

Theoabw committed:
Made edit form for info (#461)

* Made edit form for info

* fixed imports & email validation

* made email non-editable, name & last name required

Theoabw committed:
update to ckeditor 5 (#519)

* update to ckeditor5

* ckeditor5 config

* remove old config

* update ckeditor syles, use extends config

* made extends config default

jaspnas committed:
Make migrations

jaspnas committed:
Change colors
